test/router-options.test.js

```diff
@@ -78,20 +78,21 @@ test('Should honor maxParamLength option', t => {
     method: 'GET',
     url: '/test/123456789'
   }, (error, res) => {
-    t.error(error)
-    t.equal(res.statusCode, 200)
+    t.assert.ifError(error)
+    t.assert.strictEqual(res.statusCode, 200)
   })
 
   fastify.inject({
     method: 'GET',
     url: '/test/123456789abcd'
   }, (error, res) => {
-    t.error(error)
-    t.equal(res.statusCode, 404)
+    t.assert.ifError(error)
+    t.assert.strictEqual(res.statusCode, 404)
+    done()
```

Comment:
when there are more that one `inject`, we can't use the `done()` cb in a single one because it is not guaranteed that this cb will be the last one, so the test may completes before the other assertions

Reply:
I have refactored it, see commit here: [b4957ed](https://github.com/fastify/fastify/pull/5840/commits/b4957ed1902183d42c50db8c94d6e01903282e47)